resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/ServerResponseWriter.java

```diff
@@ -78,7 +78,8 @@ public static void writeNomapResponse(BuiltResponse jaxrsResponse, final HttpReq
       
       executeFilters(jaxrsResponse, request, response, providerFactory, method);
 
-      if (jaxrsResponse.getEntity() == null)
+      //[RESTEASY-1627] check on response.getOutputStream() to avoid resteasy-netty4 trying building a chunked response body for HEAD requests 
+      if (jaxrsResponse.getEntity() == null || response.getOutputStream() == null)

```

Comment:
That's what we had before RESTEASY-1365 was implemented. However, the problem was that causing Content-Length to always be set to 0 for responses to HEAD requests, which goes against rfc7230/7231. Basically, either the Content-Length is not specified at all, or it has to have the same value that would returned for a GET request to the same resource. In order to compute such a value, the response processing has to be performed and the the headers set after that, hence the removal of the check on the httpmethod name there.

The idea with the fix here is to allow resteasy-netty4 to reply to HEAD request without setting the Content-Length header, while leaving the fix for other servers as it was implemented in RESTEASY-1365 (so a valid Content-Length is computed and set).